build-tools/scripts/XAVersionInfo.targets

```diff
@@ -68,7 +68,14 @@
         ToolExe="$(GitToolExe)">
       <Output TaskParameter="AbbreviatedCommitHash"   PropertyName="XAVersionHash"          Condition=" '$(XAVersionHash)' == '' " />
     </GitCommitHash>
+    <PropertyGroup>
+      <_GitHead>$([System.IO.Path]::Combine($(XamarinAndroidSourcePath), '.git', 'HEAD'))</_GitHead>
+      <_GitHead>$([System.IO.Path]::GetFullPath($(_GitHead)))</_GitHead>
+      <_GitHead>$([System.IO.File]::ReadAllText('$(_GitHead)'))</_GitHead>
+      <XAVersionBranch>$([System.Text.RegularExpressions.Regex]::Match($(_GitHead), '(?&lt;=refs/heads/).+$'))</XAVersionBranch>
```

Comment:
Since this requires an inline `Regex` and lots of `System.IO`... Should we just put this logic in the `<GitBranch/>` task?

It would probably be a bit more readable from C#. Instead of XML/MSBuild-ese.

Reply:
@jonathanpeppers I've moved this into the GitBranch task as requested. We'll now only execute the base `ToolTask` if the `.git/HEAD` file regex matching does not work.